src/components/App/index.tsx

```diff
@@ -74,18 +80,28 @@ export default class App extends Component<Props, State> {
   }
 
   @bind
-  private onBack() {
-    this.setState({ file: undefined });
+  private onPopState() {
+    this.setState({ isEditorOpen: history.pathname === ROUTE_EDITOR });
+  }
+
+  @bind
+  private openEditor() {
+    history.push(ROUTE_EDITOR);
+    this.setState({ isEditorOpen: true });
+  }
+
+  componentDidMount() {
+    history.addPopStateListener(this.onPopState);
```

Comment:
This could just be in the constructor, right?

Reply:
done 👍